### PR TITLE
#6207 Added Random Number generator

### LINKED ISSUE #6207: Add a Pebble function to generate a random number
### Feature description

sth like in Python:

```Python
import random

random_value = random.randint(0, 1)
```

### PR DESCRIPTION
**Pull request by coderkill**
### What changes are being made and why?

Python:

import random

random_value = random.randint(0, 1)

Added that as pebble function for Java

### How the changes have been QAed?

Written extensive Junits

Closes https://github.com/kestra-io/kestra/issues/6207.

### CODE REVIEW

**Comment by MilosPaunovic** [comment]
Just a small note for the future, if you could use **GitHub Keywords** when opening PRs to connect it to the proper issue (to automate the entire process a bit more), it would be super helpful to us. You can read a bit more about it [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) (I've added closing keyword for this particular issue).

**Comment by coderkill** [comment]
Noted, will do from the next PR, Thanks for the tip!

File: core/src/main/java/io/kestra/core/runners/pebble/functions/RandomNumberGenerator.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Function;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+public class RandomNumberGenerator implements Function {
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/pebble/functions/RandomNumberGenerator.java]:
```suggestion
public class RandomNumberFunction implements Function {
```

Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/runners/pebble/functions/RandomNumberGenerator.java]: We always ends Function class with `Function`

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Comment by loicmathieu** [comment]
I think it can even be named `RandomFunction` and bind to `random()`.

I wonder if we would need a random float at some time, WDYT @anna-geller ?

File: core/src/main/java/io/kestra/core/runners/pebble/functions/RandomNumberGenerator.java (hunk 1 above)
Comment (coderkill) [reply core/src/main/java/io/kestra/core/runners/pebble/functions/RandomNumberGenerator.java]: Noted! will make change accordingly

**Comment by anna-geller** [comment]
@loicmathieu no use case for float for now, but given it's named `randomNumber()`, we can add randomFloat if needed later

we need this one for use cases like https://github.com/kestra-io/kestra/issues/5983#issuecomment-2508030834

 LGTM, we can merge as is!

**Review by anna-geller — APPROVED** [review]
thanks for adding this!

**Comment by loicmathieu** [comment]
@anna-geller well, a float is a number that's why I asked, maybe rename it `randomInt()` ?

**Comment by coderkill** [comment]
> @anna-geller well, a float is a number that's why I asked, maybe rename it `randomInt()` ?

Do you want me to make this change?

**Comment by anna-geller** [comment]
np, `randomInt()` works too!

**Review by loicmathieu — APPROVED** [review]
LGMT, thanks a lot

**Pull request merged**